examples/dial/example_dial.cpp

```diff
@@ -33,28 +34,37 @@ int main() {
     boost::optional<Viam::SDK::DialOptions> opts(dial_options);
     std::string address(uri);
     Options options = Options(1, opts);
+
+    // connect to robot, ensure we can refresh it
     std::shared_ptr<RobotClient> robot = RobotClient::at_address(address, options);
     robot->refresh();
+
+    // ensure we can query resources
     std::vector<ResourceName>* resource_names = robot->resource_names();
-    ResourceName the_one_we_care_about = resource_names->at(0);
     for (ResourceName resource : *resource_names) {
         std::cout << "Resource name: " << resource.name() << resource.type() << resource.subtype()
                   << std::endl;
     }
+
+    // ensure we can query statuses
     std::vector<Status> status_plural = robot->get_status();
     std::cout << "Status plural len " << status_plural.size() << std::endl;
     for (Status s : status_plural) {
         std::cout << " Status! " << s.name().subtype() << std::endl;
     }
 
-    std::vector<ResourceName> just_one = std::vector<ResourceName>();
-    just_one.push_back(the_one_we_care_about);
+    // ensure we can send requests for specific resources
+    std::vector<ResourceName> just_one = {resource_names->at(0)};
```

Comment:
Just wondering, is there a particular reason you are using `at()` vs the `[]` operator?

I haven't seen `at()` very much, so i'm intrigued.

Reply:
Ok :) my minor suggestion is to use `[]` - but of course there are tradeoffs! And they both do what you're wanting.

I am not too familiar with `at()`, I read through this [SO question](https://stackoverflow.com/questions/9376049/vectorat-vs-vectoroperator) and it gave me some bearings. Seems like `at()` is a little slower, but will do bounds checking for you.

I've typically seen `[]` used more in C++, in my opinion it's more preferable and idiomatic.

```suggestion
    std::vector<ResourceName> just_one = {resource_names[0]};
```

Reply:
I have generally seen `at(..)` used when the underlying type is a pointer (resouce_names is a pointer to a vector) because `(*resource_names)[0]` is uglier.

Reply:
thats a really good point. I didn't realize that thank you for clarifying, I agree with Zack in that case.